src/service/document-info-impl.js

```diff
@@ -116,6 +127,54 @@ function getPageViewId(win) {
   return String(Math.floor(win.Math.random() * 10000));
 }
 
+/**
+ * Returns an array with a total of 128 of random values based on the
+ * `win.crypto.getRandomValues` API. If that is not available concatenates
+ * a string of other values that might be hard to guess including
+ * `Math.random` and the current time.
+ * @param {!Window} win
+ * @return {!Uint8Array|string} Entropy.
+ */
+function getHighEntropy(win) {
+  // Use win.crypto.getRandomValues to get 128 bits of random value
+  const uint8array = getCryptoRandomBytesArray(win, 16); // 128 bit
+  if (uint8array) {
+    return uint8array;
+  }
+
+  // Support for legacy browsers.
+  return String(
+    win.location.href +
+      Date.now() +
+      win.Math.random() +
+      win.screen.width +
+      win.screen.height
+  );
+}
+
+/**
+ * Returns a relatively high entropy random string.
+ * This should be called once per window and then cached for subsequent
+ * access to the same value to be persistent per page.
+ * @param {!Window} win
+ * @return {!Promise<string>} pageViewId64
+ */
+function getPageViewId64(win) {
```

Comment:
Thank you @matthieuberger the implementation looks good. However I notice the code is identical to `#getNewCidForCookie ()` method [here](https://github.com/ampproject/amphtml/blob/master/src/service/cid-impl.js#L651).

To reuse code, can we rename the method to `getRandomString64()` and export the method to be used here? Thank you